proto/internalapi/common/collector_runtime_config.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+
+package common;
+
+import "google/protobuf/timestamp.proto";
+import "internalapi/common/resource_collection.proto";
+import "internalapi/common/slim-user.proto";
+
+option go_package = "common";
+
+enum RuntimeFeature {
+  PROCESSES = 0;
+  NETWORK_CONNECTIONS = 1;
+  NETWORK_ENDPOINTS = 2;
+}
+
+message Status {
+  message ProcessStatus {
+    bool enabled = 1;
+  }
+
+  message NetworkConnectionStatus {
+    bool enabled = 1;
+    bool aggregate_external = 2;
+  }
+
+  message NetworkEndpointStatus {
+    bool enabled = 1;
+    bool processes_listening_on_port = 2;
+  }
+
+  oneof status {
+    ProcessStatus process_status = 1;
+    NetworkConnectionStatus network_connection_status = 2;
+    NetworkEndpointStatus network_endpoint_status = 3;
+  }
+}
+
+message RuntimeFeatureConfig {
+  message RuntimeRule {
+    string resource_collection_name = 1;
+    Status status = 2;
+    google.protobuf.Timestamp created_at = 3;
+    google.protobuf.Timestamp last_updated = 4;
+    SlimUser created_by = 5;
+    SlimUser updated_by = 6;
+  }
+
+  RuntimeFeature feature = 1;
+  Status default_status = 2;
+  repeated RuntimeRule rules = 3;
+  google.protobuf.Timestamp created_at = 4;
+  google.protobuf.Timestamp last_updated = 5;
+  // The default is not created by a user, but by central so there is no created_by.
+  SlimUser updated_by = 6;
+}
+
+message CollectorRuntimeConfig {
+  repeated RuntimeFeatureConfig runtime_feature_config = 1;
+  repeated ResourceCollection resource_collections = 2;
+}
+
+// Collector needs to know the cluster it is running on to apply the rules correctly.
+message CollectorRuntimeConfigWithCluster {
+  string cluster_name = 1;
```

Comment:
Yes. The entire configuration is sent to each collector. It doesn't have to be that way. Sensors and collectors could just receive the parts of the configurations that are relevant to them. Given the work that has been done so far, it is probably simplest to send the entire configuration to each collector. In the future we might change it so that only the relevant parts of the configuration are sent to sensors and collectors.